test/objspace/test_objspace.rb

```diff
@@ -431,6 +431,27 @@ def dump_my_heap_please
     end
   end
 
+  def test_dump_objects_dumps_page_slot_sizes
+    assert_in_out_err(%w[-robjspace], "#{<<-"begin;"}\n#{<<-'end;'}") do |output, error|
+      begin;
+        def dump_my_heap_please
+          ObjectSpace.dump_all(output: $stdout)
+        end
+
+        p $stdout == dump_my_heap_please
+      end;
+      assert_equal 'true', output.pop
+      assert(output.count > 1)
+      output.each { |l|
+        obj = JSON.parse(l)
+        if ['CLASS', 'STRING', 'HASH'].member?(obj['type'])
```

Comment:
Why do we need this? Can we just do this?

```suggestion
        next if obj['type'] == 'ROOT'
```